code/src/pages/cart/CartPage.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from "react";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppButton from "@/components/app-button/AppButton";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import { items } from "@/pages/cart/CartPage.constants";
+import CartItem from "@/pages/cart/components/CartItem";
+import formatPrice from "@/utils/format-price/formatPrice";
+
+import "@/pages/cart/CartPage.scss";
+
+type CartItemsType = {
+  image: string;
+  name: string;
+  productPrice: number;
+  quantity: number;
+  calculatedPrice: number;
+};
+
+const CartPage = () => {
+  const [cartItems, setCartItems] = useState<CartItemsType[]>(items);
+
+  const handleQuantityChange = (index: number, newQuantity: number) => {
+    const updatedCartItems = cartItems.map((item, i) =>
+      i === index
+        ? {
+            ...item,
+            quantity: newQuantity,
+            calculatedPrice: item.productPrice * newQuantity
+          }
+        : item
+    );
+    setCartItems(updatedCartItems);
+  };
+
+  const handleDeleteItem = (index: number) => {
+    const updatedCartItems = cartItems.filter((_, i) => i !== index);
+    setCartItems(updatedCartItems);
+  };
```

Comment:
Also, replace this functions by reducers

Reply:
> Also, replace this functions by reducers

Check the description and you will find the answer
This logic will be with requests